mteb/models/vllm_wrapper.py

```diff
@@ -0,0 +1,267 @@
+            enable_prefix_caching: Whether to enable KV cache sharing for common prompt prefixes.
+                If None, uses the model's default setting.
+            gpu_memory_utilization: Target GPU memory utilization ratio (0.0 to 1.0).
+            hf_overrides: Dictionary mapping Hugging Face configuration keys to override values.
+            pooler_config: Controls the behavior of output pooling in pooling models..
+            enforce_eager: Whether to disable CUDA graph optimization and use eager execution.
+            model_prompts: A dictionary mapping task names to prompt names.
+                First priority is given to the composed prompt of task name + prompt type (query
+                or passage), then to the specific task prompt, then to the composed prompt of
+                task type + prompt type, then to the specific task type prompt, and finally to
+                the specific prompt type.
+            **kwargs: Additional arguments to pass to the vllm serving engine model.
+        """
+        requires_package(
+            self,
+            "vllm",
+            "Wrapper for vllm serving engine",
+            install_instruction="pip install mteb[vllm]",
+        )
+
+        import os
+
+        os.environ["VLLM_WORKER_MULTIPROC_METHOD"] = "spawn"
+
+        from vllm import LLM, EngineArgs
+
+        hf_overrides = {} if hf_overrides is None else hf_overrides
+
+        if head_dtype is not None:
+            hf_overrides["head_dtype"] = head_dtype
+
+        args = EngineArgs(
+            model=model,
+            revision=revision,
+            runner="pooling",
+            convert=self.convert,
+            max_model_len=max_model_len,
+            max_num_batched_tokens=max_num_batched_tokens,
+            max_num_seqs=max_num_seqs,
+            tensor_parallel_size=tensor_parallel_size,
+            data_parallel_size=data_parallel_size,
+            enable_prefix_caching=enable_prefix_caching,
+            gpu_memory_utilization=gpu_memory_utilization,
+            hf_overrides=hf_overrides,
+            pooler_config=pooler_config,
+            enforce_eager=enforce_eager,
+            trust_remote_code=trust_remote_code,
+            dtype=dtype,
+            **kwargs,
+        )
+        self.llm = LLM(**vars(args))
+        self.model_prompts = model_prompts
+
+        self.mteb_model_meta = ModelMeta.from_hub(model=model, revision=revision)
+
+        if (
+            self.model_prompts
+            and len(self.model_prompts) <= 2
+            and (
+                PromptType.query.value not in self.model_prompts
+                or PromptType.document.value not in self.model_prompts
+            )
+        ):
+            logger.warning(
+                "Encode models use prompts most often need to be configured with at least 'query' and"
+                f" 'document' prompts to ensure optimal performance. Received {self.model_prompts}"
+            )
```

Comment:
Hmm I think that there are quite a few models that have no prompts (typically older or low-resource models)